src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectExecutionManagerTest.java

```diff
@@ -0,0 +1,57 @@
+package com.synopsys.integration.jenkins.detect.substeps;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.IOException;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.jenkins.detect.exception.DetectJenkinsException;
+
+public class DetectExecutionManagerTest {
+
+    private static DetectJarManager detectJarManager = new DetectJarManager(null, null, null, null);
+    private static DetectJarManager spiedDetectJarManager = Mockito.spy(detectJarManager);
+
+    @Test
+    public void testCallInterruptedException() {
+        try {
+            Mockito.doThrow(new InterruptedException()).when(spiedDetectJarManager).setUpForExecution();
+        } catch (IOException | InterruptedException e) {
+            fail("Unexpected exception thrown", e);
+        }
+
+        assertFalse(Thread.currentThread().isInterrupted());
```

Comment:
You haven't tested anything yet, but doing an assert here implies otherwise.

Reply:
Similar to tests for [SubStep in jenkins-common](https://github.com/synopsys-sig/jenkins-common/blob/master/src/test/java/com/synopsys/integration/stepworkflow/SubStepTest.java#L195), this is a "pre" check proving that the Thread isn't interrupted until the exception is thrown, which is then verified on line #32.

Reply:
That's fair, I think if we want to do that though we should include a message that indicates why the test fails, since it's failing in setup rather than execution of the underlying code.